.github/workflows/terraform-plan-apply.yml

```diff
@@ -22,8 +30,15 @@ jobs:
         run: |
           git fetch origin main:localmain
 
-          # Gets changed json files excluding the json schema
-          export FILES=$(git diff --name-only localmain | grep .json | grep -v "trial-json-schema")
+          if [ ! -z ${{ github.event.inputs.env }} ]
```

Comment:
in this var definition, you set up a default value (dev) and set required=true.
so doesnt it means that this if statement will always evaluate to 'true'?

Reply:
the logic is - if the github.event.inputs.env is populated (which is done when manual workflow_dispatch event is triggered ) then do the statement. If the  github.event.inputs.env is NOT populated (which is done when push event is triggered ) then do the statement in the else.